### PR TITLE
config: fix load config from file

### PR DESCRIPTION
**Pull request by sylzd**
### What problem does this PR solve?

close https://github.com/pingcap/tidb/issues/29523

### What is changed and how it works?
1. fix `(o *PersistOptions) Reload()` to make persistent config work

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has configuration change
- Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tidb-ansible`](https://github.com/pingcap/tidb-ansible):
- Need to cherry-pick to the release branch

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by sylzd** [comment]
It's confused when do we use the configures of file or etcd?

**Comment by rleungx** [comment]
> It's confused when do we use the configures of file or etcd?

We only use config files for the first time starting the PD.

**Comment by sylzd** [comment]
> > It's confused when do we use the configures of file or etcd?
> 
> We only use config files for the first time starting the PD.

Got~  So https://github.com/pingcap/tidb/issues/29523 is not a bug, since "`edit-config` and `reload`" operation is not the first time to start pd. @seiya-annie

File: server/server.go [hunk 1]
```diff
@@ -498,6 +498,7 @@ func (s *Server) Run() error {
 		return err
 	}
 
+	s.initConfig()
```
Comment (nolouch) [inline server/server.go:501]: Will this overwrite the updated configuration in etcd throught `pd-ctl`?

Comment (sylzd) [reply server/server.go:501]: yep, I think I should close this pr, since  https://github.com/pingcap/tidb/issues/29523 is not a bug

**Pull request closed without merging**